Content.Goobstation.Shared/Weapons/Multishot/SharedMultishotSystem.cs

```diff
@@ -30,104 +45,139 @@ public override void Initialize()
         SubscribeLocalEvent<MultishotComponent, GotEquippedHandEvent>(OnEquipWeapon);
         SubscribeLocalEvent<MultishotComponent, GotUnequippedHandEvent>(OnUnequipWeapon);
         SubscribeLocalEvent<MultishotComponent, GunRefreshModifiersEvent>(OnRefreshModifiers);
-        SubscribeLocalEvent<MultishotComponent, GunShotEvent>(OnShotAttempt);
+        SubscribeLocalEvent<MultishotComponent, GunShotEvent>(OnGunShot);
+        SubscribeAllEvent<RequestShootEvent>(OnRequestShoot);
     }
 
-    private void OnShotAttempt(Entity<MultishotComponent> multishotWeapon, ref GunShotEvent args)
+    private void OnRequestShoot(RequestShootEvent msg, EntitySessionEventArgs args)
     {
-        var comp = multishotWeapon.Comp;
+        var user = args.SenderSession.AttachedEntity;
 
-        if (comp.RelatedWeapon == null)
+        if (user == null ||
+            !_combatSystem.IsInCombatMode(user))
             return;
 
-        // Need to prevent recursive shoot attempting
-        if (_handsSystem.GetActiveItem(args.User) != multishotWeapon)
+        var gunsEnumerator = GetMultishotGuns(user.Value);
+        var shootCoords = GetCoordinates(msg.Coordinates);
+        var target = GetEntity(msg.Target);
+
+        foreach(var gun in gunsEnumerator)
+        {
+            var (gunEnt, gunComp, multiComp) = gun;
+
+            var potentialTarget = target;
+            if (gunComp.Target == null || !gunComp.BurstActivated || !gunComp.LockOnTargetBurst)
+                gunComp.Target = potentialTarget;
+
+            _gunSystem.AttemptShoot(user.Value, gunEnt, gunComp, shootCoords);
+        }
+    }
+
+    private void OnGunShot(Entity<MultishotComponent> multishotWeapon, ref GunShotEvent args)
+    {
+        var (uid, comp) = multishotWeapon;
+
+        if (!comp.MultishotAffected)
             return;
 
-        if (!TryComp<GunComponent>(comp.RelatedWeapon, out var relatedGun) || !TryComp<GunComponent>(multishotWeapon, out var gun))
+        DamageHands(uid, comp, args.User);
+        DealStaminaDamage(uid, comp, args.User);
+    }
+
+    private void DealStaminaDamage(EntityUid weapon, MultishotComponent component, EntityUid target)
+    {
+        if (component.StaminaDamage == 0)
+            return;
+
+        _staminaSystem.TakeStaminaDamage(target, component.StaminaDamage, source: target, with: weapon, visual: false);
+    }
+
+    private void DamageHands(EntityUid weapon, MultishotComponent component, EntityUid target)
+    {
+        if (component.HandDamage == 0)
             return;
 
-        if (gun.ShootCoordinates == null)
+        if (!_handsSystem.IsHolding(target, weapon, out var hand))
             return;
 
-        if (TryComp(comp.RelatedWeapon.Value, out GunComponent? otherGun))
-            otherGun.Target = gun.Target;
+        // I didn't found better way to get hand
+        var bodySymmetry = BodyPartSymmetry.None;
```

Comment:
```suggestion
        var bodySymmetry = hand.Location switch
        {
            HandLocation.Left => BodyPartSymmetry.Left,
            HandLocation.Right => BodyPartSymmetry.Right,
            _ => BodyPartSymmetry.None,
        };
```

idk if this is better in your eyes just a suggestion, might be less readable tho but it makes sense in my mind